tools/processresults.py

```diff
@@ -0,0 +1,306 @@
+    Also modifies the input results to canonicalize all questions.
+
+  Raises:
+    QuestionError: A question in some response didn't match the expected
+        canonical version of the question
+    UnexpectedFormatError: Either results has no responses or all
+        responses use PLACEHOLDER as the text for all questions
+  """
+  # Find responses that didn't use 'PLACEHOLDER' as the text for every question
+  results[:] = [r for r in results
+                      if r['responses'][0]['question'] != 'PLACEHOLDER']
+  if not results:
+    raise UnexpectedFormatException('No results with questions found')
+
+  _ReorderAttributeQuestions(results)
+
+  _ReplaceUrlWithPlaceholder(results)
+
+  # Any response with the max number of questions should now be fine as
+  # the canonical list of questions; find one such response.
+  max_list_len = len(results[0]['responses'])
+  canonical_index = 0
+  for i, r in enumerate(results):
+    if len(r['responses']) > max_list_len:
+      max_list_len = len(r['responses'])
+      canonical_index = i
+
+  # Do some light error checking; all questions lists should now be the same,
+  # except some question lists may have extra questions. So, check each list
+  # of questions against the canonical list of questions.
+  for i, qa_pair in enumerate(
+      results[canonical_index]['responses']):
+    for j, r in enumerate(results):
+      if i < len(r['responses']):
+        if qa_pair['question'] != r['responses'][i]['question']:
+          raise QuestionError(
+              'Question text differs: Result %d question %d is {%s}. Result '
+              '%d question %d is {%s}.'
+              % (canonical_index, i, qa_pair['question'],
+                 j, i, r['responses'][i]['question']))
+
+  return canonical_index
+
+def _WriteToCsv(results, canonical_index, out_file):
+  """Write results for a given condition to a CSV file
+
+  Given a list of results in the expected format, writes
+  date_received, date_taken,  participant_id, survey_type, and
+  answers to each survey question to a CSV file. Uses
+  canonical_questions as column headers for corresponding answers.
+    
+  Args:
+    results: List of results, where each result is a dict
+        containing metadata and response data from a survey response.
+        The results list should be filtered for one condition, and
+        questions in the response data should be canonicalized.
+    canonical_index: Index of element in results whose questions can
+        be considered canonical for use as CSV column headers.
+    out_file: CSV file to write output to.
+
+  Returns:
+    None.
+  """
+  canonical_questions = [
+      qa_pair['question']
+      for qa_pair in results[canonical_index]['responses']]
+
+  # Use the Python csv library to write to csv format. The writer in this
+  # library requires data to be a list of rows in flat field_name:data form.
+  # So, we put it in the required form, open the CSV file, and write the data.
```

Comment:
Fixed the comment to explain this better. The issue was more an issue with our data rather than Python's CSV writer ability.